packages/roosterjs-editor-plugins/lib/plugins/ImageSelection/ImageSelection.ts

```diff
@@ -57,6 +69,11 @@ export default class ImageSelection implements EditorPlugin {
                         if (images.length == 1) {
                             const image = images[0];
                             this.editor.select(image);
+                            this.triggerSelectionEvent(
+                                this.editor,
+                                SelectionRangeTypes.ImageSelection,
+                                image
+                            );
                         }
                         this.imageId = null;
```

Comment:
Instead of update imageId here, let's handle the Selection Changed event in this plugin as well, and check if it is a Image Selection so everything fall into the same code path